releasenotes/notes/improve-controlled-UnitaryGate-5f3aa3d44034cfa7.yaml

```
@@ -0,0 +1,9 @@
+---
+upgrade_circuits:
+  - |
+    Previously the `control` method of UnitaryGate was utilizing `Isometry` for decomposition.
+    This improvement switches the synthesis to `qs_decomposition` which approximately halves the gate count.
+upgrade_synthesis:
+  - |
+    An improvement to the Quantum Shannon Decomposition `qs_decomposition` was added which reduces the CNOT gate count
+    in the special case that the input unitary appears to be a controlled unitary.
```

I am not sure, but it seems that this should be a new feature and not an upgrade -- unless you think that someone would want to call the `qs_decomposition` code without the new "block diagonal" optimization.

```suggestion
    The Quantum Shannon Decomposition (:func:`.qs_decomposition`) includes an optimization that reduces the CNOT gate count in the case that the input unitary happens to be a controlled unitary.
```

---

done.